### PR TITLE
Straighten out the logic around excluded pods

### PR DESCRIPTION
**Pull request by rudle**
Add a test that ought to have appeared in the original PR

### CODE REVIEW

File: bin/p2-shutdown/main.go [hunk 1]
```diff
@@ -59,12 +56,12 @@ func main() {
 	var haltWG sync.WaitGroup
 	for _, realityEntry := range reality {
 		pod := podFactory.NewLegacyPod(realityEntry.Manifest.ID())
-		if *dryRun {
-			log.Printf("dry run, skipping this pod: %s", pod.Id)
+		if !shutdownPod(pod.Id, podsToShutdown, *excludePods) {
```
Comment (mpuncel) [inline bin/p2-shutdown/main.go]: nit: `shouldShutdownPod()` would better indicate that this function doesn't perform the down shutting

File: bin/p2-shutdown/main_test.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+func TestShutdownPod(t *testing.T) {
+	testCases := []struct {
+		pod            types.PodID
+		podsToShutdown []types.PodID
+		podsToExclude  []string
+		expectation    bool
+	}{
+		{types.PodID("a"), []types.PodID{"a"}, []string{}, true},
+		{types.PodID("a"), []types.PodID{}, []string{"a"}, false},
+		{types.PodID("a"), []types.PodID{}, []string{}, true},
```
Comment (mpuncel) [inline bin/p2-shutdown/main_test.go]: maybe add a case where "b" is in teh shutdown list which should make the expectation for "a" false right?

**Comment by rudle** [comment]
Feedback addressed. 

I guess I have different git authors on my different machines. That's annoying.

File: bin/p2-shutdown/main.go [hunk 3]
```diff
@@ -84,23 +86,22 @@ func main() {
...
 	}
-	return false
-}
 
-func listInclude(pod string, pods []string) bool {
-	for _, p := range pods {
-		if p == pod {
-			return true
-		}
+	if len(podsToShutdown) > 0 {
```
Comment (petertseng) [inline bin/p2-shutdown/main.go]: I guess if you wanted to code golf you could `return len(podsToShutdown) == 0`

shrug

**Pull request merged**